elementary/monitor/api/alerts/alert_filters.py

```diff
@@ -16,6 +16,80 @@
 logger = get_logger(__name__)
 
 
+def get_string_ends(input_string: Optional[str], splitter: str) -> List[str]:
+    if input_string is None:
+        return []
+    parts = input_string.split(splitter)
+    result = []
+
+    for i in range(len(parts)):
+        result.append(splitter.join(parts[i:]))
+
+    return result
+
+
+def _get_alert_node_name(alert: PendingAlertSchema) -> Optional[str]:
+    alert_node_name = None
+    alert_type = AlertTypes(alert.type)
+    if alert_type is AlertTypes.TEST:
+        alert_node_name = alert.data.test_name  # type: ignore[union-attr]
```

Comment:
i think you can remove this type ignore by adding type hints in line 32: `alert_node_name: Optional[str] = None`

Reply:
Apparently not :( 
```
def _get_alert_node_name(alert: PendingAlertSchema) -> Optional[str]:
    alert_node_name: Optional[str] = None
    alert_type = AlertTypes(alert.type)
    if alert_type is AlertTypes.TEST:
        alert_node_name = alert.data.test_name
```

```
elementary/monitor/api/alerts/alert_filters.py:31: error: Item "ModelAlertDataSchema" of "Union[TestAlertDataSchema, ModelAlertDataSchema, SourceFreshnessAlertDataSchema]" has no attribute "test_name"  [union-attr]
elementary/monitor/api/alerts/alert_filters.py:31: error: Item "SourceFreshnessAlertDataSchema" of "Union[TestAlertDataSchema, ModelAlertDataSchema, SourceFreshnessAlertDataSchema]" has no attribute "test_name"  [union-attr]
```

It's from the [earlier version](https://github.com/elementary-data/elementary/pull/1768/files#diff-6f78902926e4d87f966d446498d442d9c2f33492171548b1a8fc72b25e49727bL153).